### PR TITLE
Refactor tests/foreman/ui/test_rhcloud_insights.py

### PR DESCRIPTION
**Pull request by jameerpathan111**
This PR aims to refactor tests present in `tests/foreman/ui/test_rhcloud_insights.py`.
It removes `test_delete_host_having_insights_recommendation`, `test_rh_cloud_insights_clean_statuses` and `test_insights_tab_on_host_details_page` as they are covered in existing tests and updates and parameterizes others for rhel 7, 8 and 9. 
It also removes `rhcloud_sat_host` which is unnecessary.

Depends on https://github.com/SatelliteQE/airgun/pull/940

### CODE REVIEW

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py -k test_insights_registration_with_capsule

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py -k test_rhcloud_insights_e2e

**Comment by jameerpathan111** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py

**Comment by jameerpathan111** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py -k test_insights_registration_with_capsule
env:
  ROBOTTELO_ui__record_video: true
```

**Comment by jameerpathan111** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py -k test_insights_registration_with_capsule
env:
  ROBOTTELO_ui__record_video: true
```

**Comment by jameerpathan111** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py -k test_insights_registration_with_capsule
env:
  ROBOTTELO_ui__record_video: true
```

**Comment by jameerpathan111** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py
env:
  ROBOTTELO_ui__record_video: true
```

**Comment by jameerpathan111** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py
env:
  ROBOTTELO_ui__record_video: true
```

**Comment by jameerpathan111** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py -k test_rhcloud_insights_e2e
airgun: 940
env:
  ROBOTTELO_ui__record_video: true
```

**Comment by jameerpathan111** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py
airgun: 940
env:
  ROBOTTELO_ui__record_video: true
```

**Comment by jameerpathan111** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py
airgun: 940
env:
  ROBOTTELO_ui__record_video: true
```

File: pytest_fixtures/component/rh_cloud.py [hunk 1]
```diff
@@ -53,35 +43,55 @@ def rhcloud_registered_hosts(organization_ak_setup, mod_content_hosts, rhcloud_s
...
+    module_target_sat.generate_inventory_report(org)
+    # Sync inventory status
+    module_target_sat.sync_inventory_status(org)
     yield rhel_contenthost
+    # Delete host
+    host = module_target_sat.api.Host().search(
+        query={"search": f'name={rhel_contenthost.hostname}'}
+    )[0]
+    host.delete()
+    assert not module_target_sat.api.Host().search(query={"search": f'name={host.name}'})
```
Comment (lpramuk) [inline pytest_fixtures/component/rh_cloud.py:67]: Why asserting in teardown?

File: robottelo/hosts.py [hunk 2]
```diff
@@ -2166,6 +2173,37 @@ def enroll_ad_and_configure_external_auth(self, ad_data):
...
+            handle_exception=True,
+        )
+
+    def sync_inventory_status(self, org):
+        """Perform inventory sync"""
+        inventory_sync = self.api.Organization(id=org.id).rh_cloud_inventory_sync()
+        wait_for(
+            lambda: self.api.ForemanTask()
+            .search(query={'search': f'id = {inventory_sync["task"]["id"]}'})[0]
+            .result
```
Comment (lpramuk) [inline robottelo/hosts.py:2198]: join with the line above (line limit is 99 chars)

**Review by lpramuk — APPROVED** [review]
Good job!

File: pytest_fixtures/component/rh_cloud.py (hunk 1 above)
Comment (jameerpathan111) [reply pytest_fixtures/component/rh_cloud.py:67]: good catch, I forgot to move this to test.

File: robottelo/hosts.py (hunk 2 above)
Comment (jameerpathan111) [reply robottelo/hosts.py:2198]: pre-commit is failing if I do this.

File: tests/foreman/ui/test_rhcloud_insights.py [hunk 3]
```diff
@@ -24,26 +24,36 @@
...
+from robottelo.constants import DNF_RECOMMENDATION
+from robottelo.constants import OPENSSH_RECOMMENDATION
+
+
+def create_insights_vulnerability(insights_vm):
+    """Function to create vulnerabilities that can be remediated."""
+    insights_vm.run(
+        'chmod 777 /etc/ssh/sshd_config;dnf update -y dnf;'
+        'sed -i -e "/^best/d" /etc/dnf/dnf.conf;insights-client'
+    )
```
[the remark this answers is not in the document]
Comment (jameerpathan111) [reply tests/foreman/ui/test_rhcloud_insights.py:36]: I'd like to be able to use it multiple times during the test that's why it's not a fixture.

File: robottelo/hosts.py [hunk 4]
```diff
@@ -1597,7 +1598,7 @@ def get_features(self):
         """Get capsule features"""
         return requests.get(f'https://{self.hostname}:9090/features', verify=False).text
 
-    def capsule_setup(self, sat_host=None, **installer_kwargs):
+    def capsule_setup(self, sat_host=None, enable_registration=False, **installer_kwargs):
```
[the remark this answers is not in the document]
Comment (jameerpathan111) [reply robottelo/hosts.py]: updated the pr.

File: pytest_fixtures/component/rh_cloud.py [hunk 5]
```diff
@@ -1,32 +1,22 @@
...
-    org = rhcloud_sat_host.api.Organization().create()
-    rhcloud_sat_host.upload_manifest(org.id, module_sca_manifest.content)
+    org = module_target_sat.api.Organization().create()
+    module_target_sat.upload_manifest(org.id, module_extra_rhel_entitlement_manifest.content)
     return org
 
 
 @pytest.fixture(scope='module')
-def organization_ak_setup(rhcloud_sat_host, rhcloud_manifest_org):
+def organization_ak_setup(module_target_sat, rhcloud_manifest_org):
```
[the remark this answers is not in the document]
Comment (jameerpathan111) [reply pytest_fixtures/component/rh_cloud.py]: updated

**Comment by jameerpathan111** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_insights.py
airgun: 940
env:
  ROBOTTELO_ui__record_video: true
```

File: tests/foreman/ui/test_rhcloud_insights.py [hunk 6]
```diff
@@ -426,87 +313,34 @@ def test_delete_host_having_insights_recommendation(
...
+        recommendations = session.host.read_insights_recommendations(rhel_insights_vm.hostname)
+        assert len(recommendations), 'No recommendations were found'
+        assert recommendations[0]['Hostname'] == rhel_insights_vm.hostname
+        assert int(result['Recommendations']) == len(recommendations)
+        # Delete host
+        host = module_target_sat.api.Host().search(
+            query={"search": f'name={rhel_insights_vm.hostname}'}
+        )[0]
+        host.delete()
+        assert not module_target_sat.api.Host().search(query={"search": f'name={host.name}'})
```
[the remark this answers is not in the document]
Comment (ogajduse) [reply tests/foreman/ui/test_rhcloud_insights.py]: Oooops, `assert not rhel_insights_vm` should have been `assert not rhel_insights_vm.nailgun_host`.
Could you please modify it?

**Review by ogajduse — APPROVED** [review]
One non-blocking comment left.

**Pull request merged**